content/en/docs/concepts/services-networking/endpoint-slices.md

```diff
@@ -58,36 +73,94 @@ endpoints:
       topology.kubernetes.io/zone: us-west2-a
 ```
 
-By default, Endpoint Slices managed by the EndpointSlice controller will have no
+By default, Endpoint Slices managed by the Endpoint Slice controller will have no
 more than 100 endpoints each. Below this scale, Endpoint Slices should map 1:1
 with Endpoints and Services and have similar performance.
 
 Endpoint Slices can act as the source of truth for kube-proxy when it comes to
 how to route internal traffic. When enabled, they should provide a performance
 improvement for services with large numbers of endpoints.
 
-## Address Types
+### Address Types
 
-EndpointSlices support three address types:
+Endpoint Slices support three address types:
 
 * IPv4
 * IPv6
 * FQDN (Fully Qualified Domain Name)
 
-## Motivation
-
-The Endpoints API has provided a simple and straightforward way of
-tracking network endpoints in Kubernetes. Unfortunately as Kubernetes clusters
-and Services have gotten larger, limitations of that API became more visible.
-Most notably, those included challenges with scaling to larger numbers of
-network endpoints.
-
-Since all network endpoints for a Service were stored in a single Endpoints
-resource, those resources could get quite large. That affected the performance
-of Kubernetes components (notably the master control plane) and resulted in
-significant amounts of network traffic and processing when Endpoints changed.
-Endpoint Slices help you mitigate those issues as well as provide an extensible
-platform for additional features such as topological routing.
+### Topology
+
+With Endpoint Slices, each endpoint can contain relevant topology information.
+Endpoint Slices managed by the Endpoint Slice controller will contain the
+following information when it's available:
```

Comment:
Are these labels per-slice or per-endpoint?

(My inferred learning outcome: the reader should be clear that if 2 endpoints have _different_ topology information, that implies 2 EndpointSlice resources in the API server).

Reply:
I would reference https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/ to clarify that this topology information is conveyed using _labels_, and that these names are meaningful outside of the context of Services and their endpoints.

Reply:
I've updated this to clarify the topology information is stored per endpoint, as well as providing more information about the source of these values. These values aren't all derived from labels, and they also aren't necessarily labels, although the validation is similar. Do the updates make sense? Is there a good place to reference the labels, annotations, and taints doc you suggested above?